docs/shared/tools-workspace-schematics.md

```diff
@@ -1 +1,284 @@
 # Using Workspace Schematics
+
+Workspace schematics provide a way to automate many tasks you regularly perform as part of your development workflow. Whether its scaffolding out components, features, or ensuring libraries are generated and structured in a certain way, schematics help you standardize these tasks in a consistent, and predictable way. Nx provides tooling around creating, and running custom schematics from within your workspace. This guide shows you how to create, run, and customize workspace schematics within your Nx workspace.
+
+## Creating a workspace schematic
+
+Use the Nx CLI to generate the initial files needed for your workspace schematic.
+
+```sh
+nx generate @nrwl/workspace:workspace-schematic my-schematic
+```
+
+After the command is finished, the workspace schematic is created under the `tools/schematics` folder.
+
+```treeview
+happynrwl/
+├── apps/
+├── libs/
+├── tools/
+│   ├── schematics
+│   |   └── my-schematic/
+│   |   |    ├── index.ts
+│   |   |    └── schema.json
+├── nx.json
+├── package.json
+└── tsconfig.json
+```
+
+The `index.ts` provides an entry point to the schematic. The file contains the factory function for the schematic to return a Rule. A rule is an operation that is performed against your filesystem.
+The `schema.json` provides a description of the schematic, available options, validation information, and default values.
+
+The initial schematic entry point contains a rule to generate a library.
+
+```ts
+import { chain, externalSchematic, Rule } from '@angular-devkit/schematics';
+
+export default function(schema: any): Rule {
+  return chain([
+    externalSchematic('@nrwl/workspace', 'lib', {
+      name: schema.name
+    })
+  ]);
+}
+```
+
+The `chain` function takes a an array of rules and combines them into a single rule. You use this function to perform multiple operations against your workspace in a single schematic. The `externalSchematic` function allows you to call schematics provided from an installed npm package external.
```

Comment:
Is that "external" needed at the end of the sentence? I'd rather just write
_"...to call schematics provided **by an** installed npm package ~~external~~"._